applications/GeoMechanicsApplication/custom_utilities/check_utilities.cpp

```diff
@@ -71,4 +73,20 @@ std::string CheckUtilities::PrintVectorContent(const std::vector<size_t>& rVecto
     return output;
 }
 
+void CheckProperties::CheckPermeabilityProperties(size_t Dimension) const
+{
+    SetNewBounds(Bounds::InclusiveLowerAndExclusiveUpper);
+    Check(PERMEABILITY_XX);
+    if (Dimension > 1) {
+        Check(PERMEABILITY_YY);
+        Check(PERMEABILITY_XY);
+    }
+    if (Dimension > 2) {
+        Check(PERMEABILITY_ZZ);
+        Check(PERMEABILITY_YZ);
+        Check(PERMEABILITY_ZX);
+    }
+    RestorePreviousBounds();
```

Comment:
Since we only use the `RestorePreviousBounds` here, I think the following would simplify. It looks a bit less polished, but I think it's worth it, because this way we can remove the added complexity of the mutable history of types vector.
```suggestion
    const auto original_bounds_type = mRangeBoundsType;
    SetNewBounds(Bounds::InclusiveLowerAndExclusiveUpper);
    Check(PERMEABILITY_XX);
    if (Dimension > 1) {
        Check(PERMEABILITY_YY);
        Check(PERMEABILITY_XY);
    }
    if (Dimension > 2) {
        Check(PERMEABILITY_ZZ);
        Check(PERMEABILITY_YZ);
        Check(PERMEABILITY_ZX);
    }
    mRangeBoundsType = original_bounds_type;
```

Reply:
changed